src/datta_lab_to_nwb/markowitz_gillis_nature_2023/fiberphotometryinterface.py

```diff
@@ -0,0 +1,198 @@
+from hdmf.backends.hdf5.h5_utils import H5DataIO
+
+# Local
+
+
+class FiberPhotometryInterface(BaseDataInterface):
+    """Fiber Photometry  interface for markowitz_gillis_nature_2023 conversion"""
+
+    def __init__(self, file_path: str, session_uuid: str, metadata_path: str):
+        # This should load the data lazily and prepare variables you need
+        columns = (
+            "uuid",
+            "signal_dff",
+            "reference_dff",
+            "uv_reference_fit",
+            "reference_dff_fit",
+            "timestamp",
+        )
+        super().__init__(
+            file_path=file_path,
+            session_uuid=session_uuid,
+            columns=columns,
+            metadata_path=metadata_path,
+        )
+
+    def get_metadata(self) -> dict:
+        metadata = super().get_metadata()
+        session_metadata = load_dict_from_file(self.source_data["metadata_path"])
+        session_metadata = session_metadata[self.source_data["session_uuid"]]
+        metadata["NWBFile"]["session_description"] = session_metadata["session_description"]
+        metadata["NWBFile"]["session_start_time"] = session_metadata["session_start_time"]
+        metadata["NWBFile"]["identifier"] = self.source_data["session_uuid"]
+        metadata["NWBFile"]["session_id"] = self.source_data["session_uuid"]
+        metadata["Subject"] = {}
+        metadata["Subject"]["subject_id"] = session_metadata["subject_id"]
+        metadata["FiberPhotometry"] = {}
+        metadata["FiberPhotometry"]["area"] = session_metadata["area"]
+        metadata["FiberPhotometry"]["reference_max"] = session_metadata["reference_max"]
+        metadata["FiberPhotometry"]["signal_max"] = session_metadata["signal_max"]
+        metadata["FiberPhotometry"]["signal_reference_corr"] = session_metadata["signal_reference_corr"]
+        metadata["FiberPhotometry"]["snr"] = session_metadata["snr"]
+
+        return metadata
+
+    def get_metadata_schema(self) -> dict:
+        metadata_schema = super().get_metadata_schema()
+        metadata_schema["properties"]["FiberPhotometry"] = {
+            "type": "object",
+            "properties": {
+                "area": {"type": "string"},
+                "reference_max": {"type": "number"},
+                "signal_max": {"type": "number"},
+                "signal_reference_corr": {"type": "number"},
+                "snr": {"type": "number"},
+            },
+        }
+        return metadata_schema
+
+    def run_conversion(self, nwbfile: NWBFile, metadata: dict):
+        """Run conversion of data from the source file into the nwbfile."""
+        session_df = pd.read_parquet(
+            self.source_data["file_path"],
+            columns=self.source_data["columns"],
+            filters=[("uuid", "==", self.source_data["session_uuid"])],
+        )
+
+        # Excitation Sources Table
+        excitation_sources_table = ExcitationSourcesTable(
+            description="ExcitationSourcesTable",
```

Comment:
I'd at least make this a human-readable description of what this table is or represents; it's not a very common extension so a brief summary might be useful